roles/core/pxe_stack/templates/nodes_parameters.yml.j2

```
@@ -19,15 +30,14 @@
   {% if hostvars[host]['equipment_profile']['equipment_type'] == "server" %}
     {% if hostvars[host]['network_interfaces'] is defined and hostvars[host]['network_interfaces'] is not none %}
 {{ host }}:
-  equipment_profile: {{ hostvars[host]['group_names'] | select('match','^'+equipment_naming+'_.*') | list | unique | sort | first | join | replace(equipment_naming + '_','') | trim }}
+  equipment_profile: {{ hostvars[host]['group_names'] | select('match','^'+equipment_naming+'_.*') | list | unique | sort | first | replace(equipment_naming + '_','') | trim }}
   network:
-{% set host_node_main_network = (macro_network.node_main_network(hostvars[host]['network_interfaces'],j2_current_iceberg_network)| trim) %}
-{% set host_node_main_network_interface = (macro_network.node_main_network_interface(hostvars[host]['network_interfaces'],host_node_main_network) | trim) %}
-    node_main_network_interface: {{ host_node_main_network_interface }}
+{% set host_node_main_network = node_main_network(hostvars[host]['network_interfaces'],j2_current_iceberg_network) | trim %}
+    node_main_network_interface: {{ node_main_network_interface(hostvars[host]['network_interfaces'],j2_current_iceberg_network) | trim }}
     node_main_network_netmask: {{ networks[host_node_main_network]['netmask'] }}
     node_main_network_gateway: {% if networks[host_node_main_network]['gateway'] is defined and not none %}{{ networks[host_node_main_network]['gateway'] }}{% endif %}
 
-    node_main_network_interface_ip: {{ hostvars[host]['network_interfaces'][host_node_main_network_interface]['ip4'] }}
+    node_main_network_interface_ip: {{ node_main_network_ip4(hostvars[host]['network_interfaces'],j2_current_iceberg_network) | trim }}
```

Following the macro change above:
```suggestion
    node_main_network_interface_ip: {{ host_node_main_network_interface['ip4'] | trim }}
```

Code untested, just a thought.

---

Regarding j2_variables.

The issue related to presence of j2_variables into main inventory will be solved in a next PR, still for 1.3. I preferred to not modify everything into this one :-)

Put simply:

```
mkdir /etc/bluebanquise/internal/group_vars/all/
mv /etc/bluebanquise/resources/example/simple_cluster/inventory/group_vars/all/j2_variables /etc/bluebanquise/internal/group_vars/all/
```

Leads to:

```
[root@localhost bluebanquise]# tree internal/
internal/
└── group_vars
    └── all
        ├── internal_variables.yml
        ├── j2_variables
        │   ├── groups.yml
        │   ├── icebergs.yml
        │   ├── network.yml
        │   └── README.md
        └── README.md

3 directories, 6 files
[root@localhost bluebanquise]#
```

Then in ansible.cfg:

`inventory      = inventory,internal`

At execution, both inventories are gathered.
This will simplify j2_variables usage, update, etc, without impacting main inventory.

If need manual execution with custom inventory, users can use double -i on ansible-playbook or ANSIBLE_INVENTORY variable with comma separated.

---

I finished reading comments. Thanks for these 😊 
I will update the PR asap.